private/bufpkg/bufplugin/bufplugindocker/docker.go

```diff
@@ -225,6 +225,11 @@ func NewClient(logger *zap.Logger, cliVersion string, options ...ClientOption) (
 		client.WithHTTPHeaders(map[string]string{
 			"User-Agent": BufUpstreamClientUserAgentPrefix + cliVersion,
 		}),
+		// NOTE: This setting is not safe when a client is called from multiple goroutines:
```

Comment:
This feels like a massive potential bug in the future - are we sure we want to do this?

Reply:
right now it just breaks when used against any server that is a lower version, not sure what our other mitigations are, the docker team [are aware of this](https://github.com/moby/moby/issues/43729#issuecomment-1174446640) so will hopefully fix in the near future
```
Error response from daemon: client version 1.42 is too new. Maximum supported API version is 1.41
```

Reply:
We are only using this to push custom plugins - there are no other code paths that use these Docker abstractions and no concurrency in play there.

The alternative is that we hardcode an earlier version of the Docker API version that is compatible with most systems.

Reply:
PTAL - I updated to force version negotiation when creating the client so reads/writes to those fields should no longer race.